src/main/java/com/flansmod/client/ClientRenderHooks.java

```diff
@@ -582,12 +582,21 @@ public void renderPlayer(RenderPlayerEvent.Pre event)
 		}
 	}
 	
-	public void cameraSetup(CameraSetup event)
+public void cameraSetup(CameraSetup event)
 	{
 		if(mc.thePlayer.ridingEntity instanceof IControllable)
 		{
 			IControllable cont = (IControllable)mc.thePlayer.ridingEntity;
-			event.roll = interpolateRotation(cont.getPrevPlayerRoll(), cont.getPlayerRoll(), (float)event.renderPartialTicks);
+			float roll = interpolateRotation(cont.getPrevPlayerRoll(), cont.getPlayerRoll(), (float)event.renderPartialTicks);
+			if(((EntitySeat)mc.thePlayer.ridingEntity).driveable != null){
+			EntityDriveable ent = ((EntitySeat)mc.thePlayer.ridingEntity).driveable;
+			
+			if(ent.getDriveableType().canRoll){
```

Comment:
If we can roll, we set the roll to 0? Surely, you mean if (!ent.getDriveableType().canRoll)?